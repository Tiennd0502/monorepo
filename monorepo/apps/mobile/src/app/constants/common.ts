export const CONGRATS_IMAGE = 'https://i.ibb.co/nkf6hFr/congrats.png';
