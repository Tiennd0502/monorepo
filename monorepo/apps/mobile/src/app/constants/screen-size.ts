import { Dimensions } from 'react-native';

export const BASE_SCREEN_WIDTH = 412;
export const BASE_SCREEN_HEIGHT = 842;

export const SCREEN_WIDTH: number = Dimensions.get('window').width;
export const SCREEN_HEIGHT: number = Dimensions.get('window').height;

export const CONTENT_WIDTH: number = SCREEN_WIDTH - 40;
