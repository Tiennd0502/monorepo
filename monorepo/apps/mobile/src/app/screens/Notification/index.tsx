import { useCallback } from 'react';
import { ScrollView, Stack } from 'tamagui';

// Constants
import { NOTIFICATIONS } from '@monorepo/mocks';

// Types
import { NEW_STATUS } from '@monorepo/types';

// Components
import { Notification, Divider } from '@monorepo/ui';
import { MainLayout } from '../../components';

const NotificationScreen = () => {
  const handleViewDetail = useCallback(() => null, []);

  return (
    <MainLayout padding={0} marginTop="$3">
      <ScrollView showsVerticalScrollIndicator={false}>
        {NOTIFICATIONS.map((item, index) => (
          <Stack flex={1} key={item.id}>
            <Notification {...item} onPress={handleViewDetail} />
            {item.status === NEW_STATUS.NORMAL &&
              index !== NOTIFICATIONS.length && (
                <Stack paddingHorizontal="$5">
                  <Divider marginRight="$5" color="$backgroundTertiary" />
                </Stack>
              )}
          </Stack>
        ))}
      </ScrollView>
    </MainLayout>
  );
};

export default NotificationScreen;
