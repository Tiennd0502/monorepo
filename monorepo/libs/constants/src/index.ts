export * from './apis';
export * from './message';
export * from './commons';
export * from './product';
export * from './category';
export * from './validation';
