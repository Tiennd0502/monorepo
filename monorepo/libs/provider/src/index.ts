export { default as TamaguiProvider } from './TamaguiProvider';
export { default as QueryClientProvider } from './QueryClientProvider';
