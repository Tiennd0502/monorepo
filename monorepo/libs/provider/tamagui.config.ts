import { createTamagui, createTokens } from 'tamagui';
import { config } from '@tamagui/config/v3';
import { themes } from '@tamagui/themes';

import {
  colorsMapping,
  radius,
  size,
  space,
  zIndex,
  defaultFont,
  headingFont,
} from '@monorepo/ui';

const tokens = {
  radius,
  size,
  space,
  zIndex,
};

const tamaguiConfig = createTamagui({
  ...config,
  defaultFont: 'body',
  themes,
  fonts: {
    body: defaultFont,
    heading: headingFont,
  },
  tokens: createTokens({
    ...tokens,
    color: colorsMapping,
  }),
});

type AppConfig = typeof config;

declare module 'tamagui' {
  type TamaguiCustomConfig = AppConfig;
}

export default tamaguiConfig;
