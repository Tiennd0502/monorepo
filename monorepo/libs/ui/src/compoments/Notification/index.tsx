import { memo, useCallback, useMemo } from 'react';
import { Stack, Image, XStack } from 'tamagui';

import { NEW_STATUS, New } from '@monorepo/types';
import Text from '../Text';

interface NotificationProps extends New {
  onPress: (id: string) => void;
}

const Notification = ({
  id,
  image,
  title,
  status,
  description,
  onPress,
}: NotificationProps) => {
  const handleClick = useCallback(() => onPress(id), [id, onPress]);

  const isStatusNormal = useMemo(() => status === NEW_STATUS.NORMAL, [status]);

  return (
    <XStack
      onPress={handleClick}
      padding="$5"
      columnGap="$3"
      position="relative"
      backgroundColor={isStatusNormal ? 'transparent' : '$backgroundTertiary'}
    >
      {image && (
        <Image
          width="$17.5"
          height="$17.5"
          borderRadius="$2"
          source={{ uri: image }}
        />
      )}
      <Stack flex={1} rowGap={5}>
        <Text
          numberOfLines={1}
          ellipsizeMode="tail"
          color="$primary"
          fontWeight="bold"
          fontSize={image ? '$3' : '$3.5'}
        >
          {title}
        </Text>
        <Text
          numberOfLines={3}
          ellipsizeMode="tail"
          textAlign="justify"
          fontSize="$2.5"
          color="$textLabel"
          lineHeight={15}
        >
          {description}
        </Text>
        {!isStatusNormal && (
          <Text
            position="absolute"
            right={0}
            bottom={-12}
            fontWeight="bold"
            color={
              status === NEW_STATUS.NEW ? '$successPrimary' : '$errorPrimary'
            }
          >
            {status}
          </Text>
        )}
      </Stack>
    </XStack>
  );
};

export default memo(Notification);
