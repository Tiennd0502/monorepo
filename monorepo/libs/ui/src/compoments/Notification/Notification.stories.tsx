import { Stack } from 'tamagui';
import { Meta, StoryObj } from '@storybook/react';

import { NOTIFICATIONS } from '@monorepo/mocks';

import Notification from '.';

const meta: Meta<typeof Notification> = {
  title: 'Components/Notification',
  component: Notification,
  decorators: [
    (Story) => (
      <Stack width={450}>
        <Story />
      </Stack>
    ),
  ],
};

export default meta;

type Story = StoryObj<typeof Notification>;

export const Default: Story = {
  args: {
    ...NOTIFICATIONS[3],
    onPress: () => null,
  },
};

export const New: Story = {
  args: {
    ...NOTIFICATIONS[0],
    onPress: () => null,
  },
};

export const Hot: Story = {
  args: {
    ...NOTIFICATIONS[2],
    onPress: () => null,
  },
};
