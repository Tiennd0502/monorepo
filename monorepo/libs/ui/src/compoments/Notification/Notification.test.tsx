import {fireEvent, render, waitFor} from '@testing-library/react-native';

import { TamaguiProvider } from '@monorepo/provider';
import { NOTIFICATIONS } from '@monorepo/constants';
import Notification from '.';

const onPress = jest.fn();

const baseProps = { onPress };

const setup = (moreProps = {}) => {
  const props = {
    ...baseProps,
    ...moreProps,
  };
  
  return render(
    <TamaguiProvider>
      <Notification {...props} />
    </TamaguiProvider>
  );
};

describe('Notification', () => {

  const props = {
    ...NOTIFICATIONS[0],
    onPress,
  };

  it('should render Notification component', () => {
    const container = setup(NOTIFICATIONS[3]);

    expect(container).toMatchSnapshot();
  });

  it('should render Notification component status new', () => {
    const container = setup(NOTIFICATIONS[0]);

    expect(container).toMatchSnapshot();
  });

  it('should render Notification component with status hot', () => {
    const container = setup(NOTIFICATIONS[2]);

    expect(container).toMatchSnapshot();
  });

  it('calls onPress when detail clicked', () => {
    const {getByTestId} = setup(NOTIFICATIONS[0]);
    
    waitFor(() => {
      const detail = getByTestId('notification');
      fireEvent.press(detail);

      expect(onPress).toHaveBeenCalled();
    });
  });
});
