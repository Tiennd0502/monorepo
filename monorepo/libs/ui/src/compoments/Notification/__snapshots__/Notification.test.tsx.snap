// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification should render Notification component 1`] = `
<View
  style={
    {
      "backgroundColor": "#f0f0f0",
      "columnGap": 12,
      "flexDirection": "row",
      "paddingBottom": 20,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 20,
      "position": "relative",
    }
  }
  testID="notification"
>
  <Image
    source={
      {
        "uri": "https://i.ibb.co/m8nVWx2/sarah-dorweiler-fr0-J5-GIVyg-unsplash-1.png",
      }
    }
    style={
      {
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
        "height": 70,
        "width": 70,
      }
    }
  />
  <View
    style={
      {
        "flex": 1,
        "rowGap": 5,
      }
    }
  >
    <Text
      allowFontScaling={false}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        {
          "color": "#242424",
          "fontFamily": "NunitoSans, Helvetica, Arial, sans-serif",
          "fontSize": 12,
          "fontWeight": "bold",
          "lineHeight": 20,
        }
      }
      suppressHighlighting={true}
    >
      Your order #123456789 has been confirmed
    </Text>
    <Text
      allowFontScaling={false}
      ellipsizeMode="tail"
      numberOfLines={3}
      style={
        {
          "color": "#808080",
          "fontFamily": "NunitoSans_7pt-Medium",
          "fontSize": 10,
          "lineHeight": 15,
          "textAlign": "justify",
        }
      }
      suppressHighlighting={true}
    >
      Lorem, ipsum dolor sit amet consectetur adipisicing elit. Reiciendis possimus quod explicabo maiores repudiandae laudantium quam provident consectetur in dolores.
    </Text>
    <Text
      allowFontScaling={false}
      style={
        {
          "bottom": -16,
          "color": "#27ae60",
          "fontFamily": "NunitoSans, Helvetica, Arial, sans-serif",
          "fontSize": 14,
          "fontWeight": "bold",
          "lineHeight": 20,
          "position": "absolute",
          "right": 0,
        }
      }
      suppressHighlighting={true}
    >
      New
    </Text>
  </View>
</View>
`;

exports[`Notification should render Notification component status new 1`] = `
<View
  style={
    {
      "backgroundColor": "transparent",
      "columnGap": 12,
      "flexDirection": "row",
      "paddingBottom": 20,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 20,
      "position": "relative",
    }
  }
  testID="notification"
>
  <Image
    source={
      {
        "uri": "https://i.ibb.co/m8nVWx2/sarah-dorweiler-fr0-J5-GIVyg-unsplash-1.png",
      }
    }
    style={
      {
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
        "height": 70,
        "width": 70,
      }
    }
  />
  <View
    style={
      {
        "flex": 1,
        "rowGap": 5,
      }
    }
  >
    <Text
      allowFontScaling={false}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        {
          "color": "#242424",
          "fontFamily": "NunitoSans, Helvetica, Arial, sans-serif",
          "fontSize": 12,
          "fontWeight": "bold",
          "lineHeight": 20,
        }
      }
      suppressHighlighting={true}
    >
      Your order #123456789 has been canceled
    </Text>
    <Text
      allowFontScaling={false}
      ellipsizeMode="tail"
      numberOfLines={3}
      style={
        {
          "color": "#808080",
          "fontFamily": "NunitoSans_7pt-Medium",
          "fontSize": 10,
          "lineHeight": 15,
          "textAlign": "justify",
        }
      }
      suppressHighlighting={true}
    >
      Lorem, ipsum dolor sit amet consectetur adipisicing elit. Reiciendis possimus quod explicabo maiores repudiandae laudantium quam provident consectetur in dolores.
    </Text>
  </View>
</View>
`;

exports[`Notification should render Notification component with status hot 1`] = `
<View
  style={
    {
      "backgroundColor": "#f0f0f0",
      "columnGap": 12,
      "flexDirection": "row",
      "paddingBottom": 20,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 20,
      "position": "relative",
    }
  }
  testID="notification"
>
  <Image
    source={
      {
        "uri": "https://i.ibb.co/m8nVWx2/sarah-dorweiler-fr0-J5-GIVyg-unsplash-1.png",
      }
    }
    style={
      {
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
        "height": 70,
        "width": 70,
      }
    }
  />
  <View
    style={
      {
        "flex": 1,
        "rowGap": 5,
      }
    }
  >
    <Text
      allowFontScaling={false}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        {
          "color": "#242424",
          "fontFamily": "NunitoSans, Helvetica, Arial, sans-serif",
          "fontSize": 12,
          "fontWeight": "bold",
          "lineHeight": 20,
        }
      }
      suppressHighlighting={true}
    >
      Discover hot sale furnitures this week.
    </Text>
    <Text
      allowFontScaling={false}
      ellipsizeMode="tail"
      numberOfLines={3}
      style={
        {
          "color": "#808080",
          "fontFamily": "NunitoSans_7pt-Medium",
          "fontSize": 10,
          "lineHeight": 15,
          "textAlign": "justify",
        }
      }
      suppressHighlighting={true}
    >
      Lorem, ipsum dolor sit amet consectetur adipisicing elit. Reiciendis possimus quod explicabo maiores repudiandae laudantium quam provident consectetur in dolores.
    </Text>
    <Text
      allowFontScaling={false}
      style={
        {
          "bottom": -16,
          "color": "#eb5757",
          "fontFamily": "NunitoSans, Helvetica, Arial, sans-serif",
          "fontSize": 14,
          "fontWeight": "bold",
          "lineHeight": 20,
          "position": "absolute",
          "right": 0,
        }
      }
      suppressHighlighting={true}
    >
      HOT!
    </Text>
  </View>
</View>
`;
